core/src/main/java/org/apache/carbondata/core/datamap/DistributableDataMapFormat.java

```diff
@@ -158,7 +166,11 @@ public float getProgress() throws IOException, InterruptedException {
 
       @Override
       public void close() throws IOException {
-
+        if (null != dataMaps) {
+          for (DataMap dataMap : dataMaps) {
+            dataMap.clear();
```

Comment:
This impacts the datamaps which they want to cache on executor side. It always clears cache for each call.  Better add another method in datamap interface close or finish to release the resources which are need to released for every call